src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -786,6 +786,56 @@ internal HttpRequestMessage BuildSetTriggersRequest()
             }
         }
 
+        public async Task<GetTriggersResult> GetTriggersPayloadAsync()
+        {
+            var result = new GetTriggersResult
+            {
+                Success = true
+            };
+
+            if (!IsSyncTriggersEnvironment(_webHostEnvironment, _environment))
+            {
+                result.Success = false;
+                result.Error = "Invalid environment for GetTriggers operation.";
+                _logger.LogWarning(result.Error);
+                return result;
+            }
+
+            try
+            {
+                await _syncSemaphore.WaitAsync();
+
+                PrepareSyncTriggers();
+
+                var hashBlobClient = await GetHashBlobAsync();
+
+                var payload = await GetSyncTriggersPayload();
+                if (payload.Count == 0)
+                {
+                    // We don't do background sync for empty triggers.
```

Comment:
We're not doing a background sync here. This whole if block can go away

Reply:
removed the comments, but can we just print a debug log if count is 0?

Reply:
No need to do that

Reply:
ok, removed it